tests/migrations/test_autodetector.py

```diff
@@ -2361,6 +2361,29 @@ def test_multiple_bases(self):
         self.assertOperationAttributes(changes, "app", 0, 3, name="D")
         self.assertOperationAttributes(changes, "app", 0, 4, name="E")
 
+    def test_metaclass(self):
+        class AlphaBase(type):
+            pass
+
+        class Alpha(metaclass=AlphaBase):
+            pass
+
+        class ABase(AlphaBase, type(models.Model)):
+            pass
+
+        A = ModelState(
+            "app",
+            "A",
+            [("a_id", models.AutoField(primary_key=True))],
+            bases=(Alpha, models.Model),
+            metaclass=ABase
+        )
+        changes = self.get_changes([], [A])
+        # Right number/type of migrations?
+        self.assertNumberMigrations(changes, "app", 1)
+        self.assertOperationTypes(changes, "app", 0, ["CreateModel"])
+        self.assertOperationAttributes(changes, "app", 0, 0, name="A")
+
```

Comment:
Could you also test the reverse, please: `self.get_changes([A], [])`

You also have a branch in your code above (https://github.com/django/django/pull/13384/files#diff-c8a3e1d248f8f5b10af7ac5b1d8479f4R579) which raises a `TypeError` which I don't see being tested.

Reply:
@MarkusH changes have been made, ready for another look